echo-core/src/main/java/com/netflix/spinnaker/echo/artifacts/MessageArtifactTranslator.java

```diff
@@ -72,7 +82,11 @@ public MessageArtifactTranslator(InputStream templateStream, JinjavaFactory jinj
     }
 
     ObjectMapper mapper = new ObjectMapper();
-    return readArtifactList(mapper, jinjaTransform(mapper, messagePayload));
+    List<Artifact> artifacts = readArtifactList(mapper, jinjaTransform(mapper, messagePayload));
+    if (!artifacts.isEmpty()) {
+      applicationEventPublisher.publishEvent(new ArtifactEvent(null, artifacts));
```

Comment:
This is where the artifacts get sent to the `ArtifactEmitter` - via spring application events.